library/common/glfw/embedder.cc

```diff
@@ -67,41 +78,91 @@ static FlutterEmbedderState *GetSavedEmbedderState(GLFWwindow *window) {
       glfwGetWindowUserPointer(window));
 }
 
+// SyncWindowMetrics uses cached stats to send a window metrics event to
+// the Flutter Engine. It calculates the pixel_ratio based on pixel to screen coordinate
+// ratio and monitor DPI.
+// The Android/Flutter pixel_ratio is based on 160 dpi = 1.0
+static void SyncWindowMetrics(GLFWwindow *window) {
+  auto state = GetSavedEmbedderState(window);
+  double screenCoordinatesPerInch = state->monitorWidth / (state->monitorWidthMm/25.4);
+  double dpi = screenCoordinatesPerInch * state->pixelsPerScreenCoordinate;
+  double pixel_ratio = dpi / 160.0;
+
+  FlutterWindowMetricsEvent event = {};
+  event.struct_size = sizeof(event);
+  event.width = state->widthPx;
+  event.height = state->heightPx;
+  event.pixel_ratio = pixel_ratio;
+  FlutterEngineSendWindowMetricsEvent(state->engine, &event);
+}
+
+static void UpdateMonitorStats(GLFWwindow *window) {
+  auto state = GetSavedEmbedderState(window);
+
+  auto monitor = glfwGetPrimaryMonitor();
+  int monitorWidthMm, monitorHeightMm;
+  glfwGetMonitorPhysicalSize(monitor, &monitorWidthMm, &monitorHeightMm);
+  state->monitorWidthMm = monitorWidthMm;
+  state->monitorHeightMm = monitorHeightMm;
+
+  auto monitorMode = glfwGetVideoMode(monitor);
+  state->monitorWidth = monitorMode->width;
+  state->monitorHeight = monitorMode->height;
+
+  SyncWindowMetrics(window);
+}
+
+static void GLFWWindowSizeCallback(GLFWwindow *window, int width, int height) {
```

Comment:
This should be eliminated; GLFWFramebufferSizeCallback is called for resizes as well, so currently the code is updating the engine twice for every resize. You can just call `glfwGetWindowSize` in the framebuffer callback to compute the pixels-per-coordinate ratio.

Reply:
Removed